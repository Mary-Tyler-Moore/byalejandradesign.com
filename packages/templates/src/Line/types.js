/** @flow */
export type Props = {
  label: string,
  children?: React.Node,
};
