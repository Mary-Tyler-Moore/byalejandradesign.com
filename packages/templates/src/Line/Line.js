/** @flow */
import * as React from 'react';
import { camelToTitle } from 'smalldash';
// types
import type { Props } from './types';

const Line = ({ label, children }: Props) =>
  label && children ? (
    <p className="body-sourceSans-2">
      <strong>{camelToTitle(label)}: </strong>
      {children}
    </p>
  ) : null;

export default Line;
