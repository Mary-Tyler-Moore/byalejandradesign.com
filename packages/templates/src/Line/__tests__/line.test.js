import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Line from '../Line';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<Line />);
    expect(render.toJSON()).toMatchSnapshot();
  });

  test('it loads line items', () => {
    const render = TestRenderer.create(<Line label="label">content</Line>);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
