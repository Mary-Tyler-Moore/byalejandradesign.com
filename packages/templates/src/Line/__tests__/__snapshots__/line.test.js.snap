// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads line items 1`] = `
<p
  className="body-sourceSans-2"
>
  <strong>
    Label
    : 
  </strong>
  content
</p>
`;

exports[`it works it loads without crashing 1`] = `null`;
